example/example_gradient_based_opt.jl

```diff
@@ -68,6 +94,23 @@ function fdiff_all!(g, x...)
     end
 end
 
+function fdiff_all_for_constraint!(g, x...)
```

Comment:
There must be a way we can avoid this. By doing this calculation, you are doubling the number of function evaluations. If you want to impose each constraint with a separate function, it would scale proportionally to the number of constraints. With `fdiff_all!` we are already perturbing the TASOPT aircraft (and all its outputs) by all the optimization variables. Talk to Dr. Ranjan about the best way to do this. Maybe you can save the value of the constraint derivatives in some temporary file or global variable?